scipy/stats/_distribution_infrastructure.py

```diff
@@ -5219,3 +3582,288 @@ def __rtruediv__(self, other):
+    from `components` according to the probabilities given by `weights`, then
+    the random variable is realized[1]_.
+
+    Parameters
+    ----------
+    components : sequence of `ContinuousDistribution`
+        The underlying instances of `ContinuousDistribution`.
+        All must have scalar shape parameters.
+    weights : sequence of floats
+        The corresponding probabilities of selecting each random variable.
+        Must be non-negative and sum to one.
+
+    Methods
+    -------
+    support
+
+    sample
+
+    moment
+
+    mean
+    median
+    mode
+
+    variance
+    standard_deviation
+
+    skewness
+    kurtosis
+
+    pdf
+    logpdf
+
+    cdf
+    icdf
+    ccdf
+    iccdf
+
+    logcdf
+    ilogcdf
+    logccdf
+    ilogccdf
+
+    entropy
+
+    Notes
+    -----
+    The following abbreviations are used throughout the documentation.
+
+    - PDF: probability density function
+    - CDF: cumulative distribution function
+    - CCDF: complementary CDF
+    - entropy: differential entropy
+    - log-*F*: logarithm of *F* (e.g. log-CDF)
+    - inverse *F*: inverse function of *F* (e.g. inverse CDF)
+
+    References
+    ----------
+    .. [1] Mixture distribution, *Wikipedia*,
+           https://en.wikipedia.org/wiki/Mixture_distribution
+
+    """
+    # Todo:
+    # Fix Normal(mu=0.5).logentropy() runtime warning
+    # Add support for array shapes, weights
+
+    def _input_validation(self, components, weights):
+        if len(components) == 0:
+            message = ("`components` must contain at least one random variable.")
+            raise ValueError(message)
+
+        for var in components:
+            # will generalize to other kinds of distributoins when there
+            # *are* other kinds of distributions
+            if not isinstance(var, ContinuousDistribution):
+                message = ("Each element of `components` must be an instance of "
+                           "`ContinuousDistribution`.")
+                raise ValueError(message)
+            if not var._shape == ():
+                message = "All elements of `components` must have scalar shapes."
+                raise ValueError(message)
+
+        if weights is None:
+            return
+
+        weights = np.asarray(weights)
+        if len(components) != len(weights):
+            message = "`components` and `weights` must have the same length."
+            raise ValueError(message)
+
+        if not np.issubdtype(weights.dtype, np.inexact):
+            message = "`weights` must have floating point dtype."
+            raise ValueError(message)
+
+        if not np.isclose(np.sum(weights), 1.0):
+            message = "`weights` must sum to 1.0."
+            raise ValueError(message)
+
+        if not np.all(weights >= 0):
+            message = "All `weights` must be non-negative."
+            raise ValueError(message)
+
+    def __init__(self, components, /, *, weights=None):
```

Comment:
> Since I’m not sure if equal weights is a typical usage for mixture distributions

This might be true. It sounds like neither of us are sure. I'm open to this change, though, if there is more support for it.

> To me equal weights by default is like 1/2 probability by default for a Binomial distribution, which may be “fine” but would be clearer if spelled out explicitly.

I would compare it to a default value of `f_exp` in `chisquare` (which exists) or `p` for `multinomial`(which doesn't). Or perhaps even better: `weights` has a default in `gaussian_kde`. 

(That said, I assumed this would be used primarily with small numbers of terms in the PDF, hence the loop to compute the mixture PDF. If we want to support things like KDE with arbitrary kernels, it might be best to form a big array and `sum`.)

> Plus, from API maintenance perspective, it is easy to go from not accepting `None` to accepting `None`, but the converse would break backward compatibility.

I'm aware. But this same logic could be used against any function having any defaults, and is more convincing when there is concern that having a default could cause problems. Let's focus on the part that is specific to this function.

Reply:
I find your comments convincing.

Separately, speaking of mixture of a large number of random variables, I thought about the potential confusion of “mixture” distribution with “compound” distribution, i.e. one where a parameter of a distribution is drawn from a distribution. I wanted to point out in the doc that `Mixture` does not refer to a compound distribution, but failed to find a concise wording for that to avoid creating more confusion that it resolves.

Maybe if you add a `CompoundDistribution` class in the future, could add a link in the See also section.

(This brings up again the obvious question of whether `Mixture` or `MixtureDistribution` is better. I feel fine with either.)

Reply:
Great.

Actually, yeah, I'm leaning toward `MixtureDistribution` now if it's going to be a subclass of `_ProbabilityDistribution` (and someday `ProbabilityDistribution`).

Reply:
One possibility is to define a function `scipy.stats.mixture(components, weights)` that takes random variable *instances* and returns a random variable *instance* (of non-public type `_Mixture`). Looks nice to me!

Reply:
If there are no situations where someone might want to instantiate a mixture model but not care about the weights because they will be fit later, then I agree with not having a default argument for weights.

Reply:
In https://github.com/scipy/scipy/pull/21700#issuecomment-2412984394, I demonstrated how a mixture (and the underlying normal distributions) could be fit quite easily, even though there is not yet a built-in function. I happened to specify the initial weights (both equal), but could have left it unspecified. Does this count as a situation in which one would not care about the weights?

If not (so there would be no default), should `weights` still be keyword only, or should we allow positional use?

BTW, because we discussed making `components` an attribute, we must be pretty confident in that name, so I changed that first argument from positional-only to positional or keyword.

Reply:
I see. In https://github.com/scipy/scipy/pull/21700#issuecomment-2412984394, the default isn't actually used, but it seems like the idea is that when you fit with an iterative process, you need some initial guess for the weights, and the default argument is useful as the initial guess for the weights? Sure, I think that's a good enough reason to keep a default, so I think it's fine as is.